useful-scripts/importer-tests/importer.py

```diff
@@ -0,0 +1,219 @@
+        },
+        files = {
+            "zipFile": file_z
+        }
+    )
+
+
+def check_subject_status(random_identifier, API_URL, release_id):
+    response = requests.request(
+        "GET",
+        url=f'{API_URL}/api/release/{release_id}/data/subject-{random_identifier}.csv/import/status',
+        verify=False,
+        headers={
+            "Authorization": f"Bearer {os.getenv('JWT_TOKEN')}",
+            "Content-Type": "application/json",
+        },
+    )
+    if(response.json()['status'] == 'NOT_FOUND'): 
+        print('failed to find subject. Trying again to find import status')
+        check_subject_status(random_identifier, API_URL, release_id)
+
+    elif(response.json()['status'] == 'COMPLETE'):
+        subject_end_time = time.perf_counter()
+        print(f'PUBLICATION URL > {API_URL}/publication/{publication_id}/release/{release_id}/data')
+        print(f'elapsed time > ', subject_end_time - subject_start_time)
+
+    else:
+        status = response.json()['status']
+        percentageComplete = response.json()['percentageComplete']
+        print(f'subject is in stage >', status, flush=True)
+        print(f'percentage > {percentageComplete}% ', flush=True)
+        time.sleep(2)
+        # uncomment the below to send STDOUT to log file
+        # log = open(f'test-results/importer-log-{datetime.date.today()}.txt', 'a+')
+        # sys.stdout = log
+        check_subject_status(random_identifier, API_URL, release_id)
+
+
+def rename_csv_file(random_identifier):
+    try: 
+        def get_all_file_paths(directory): 
+            file_paths = [] 
+            for root, directories, files in os.walk(directory): 
+                for filename in files: 
+                    filepath = os.path.join(root, filename) 
+                    file_paths.append(filepath) 
+            return file_paths  
+
+        with zipfile.ZipFile('archive.zip', 'r') as zip_file:
+            zip_file.extractall('./test-files')
+
+        meta_filepath = glob.glob(f'./test-files/*.meta.csv')[0]
+        meta_filename = os.path.basename(meta_filepath)
+        os.rename(f'./test-files/{meta_filename}', f'./test-files/subject-{random_identifier}.meta.csv')
+
+        data_filepath = glob.glob(f'./test-files/*.csv')[0]
+        data_filename = os.path.basename(data_filepath)
+        os.rename(f'./test-files/{data_filename}', f'./test-files/subject-{random_identifier}.csv')
+
+        with zipfile.ZipFile(f'test-{random_identifier}.zip', 'w') as zip: 
+            test_dir = './test-files'
+            file_path = get_all_file_paths(test_dir)
+            for individual_file in file_path:
+                zip.write(individual_file, basename(individual_file))
+    except Exception as e:
+        print(f'Error: {e}')
+
+
+if __name__ == "__main__":
+    # To prevent InsecureRequestWarning
+    requests.packages.urllib3.disable_warnings()
+
+    load_dotenv('.env')
+    API_URL = os.getenv('API_URL')
```

Comment:
I'd maybe just lowercase API_URL (the variable name, rather than the key used for `os.getenv()`) to fit in with the general variable naming convention.  Then happy for you to merge it in!

For a future change, you might want to treat API_URL and JWT_TOKEN in the same way i.e. either look them both up here and then pass them in as parameters to methods, or don't look them up here or pass them in as method parameters, but instead have reusable `get_api_url()` and `get_jwt_token()` methods which then in turn grab them from the .env file directly.

Reply:
Cool. Still getting used to the snake_case convention that python uses haha. I've moved all references of `os.getenv()` and placed them in variables under the `load_dotenv(.env)` section. I've also created a topic_id variable which removed the instance where I was using `os.getenv('TOPIC_ID')` from the `create_publication` function